hak/matrix-seshat/check.js

```diff
@@ -61,17 +62,20 @@ module.exports = async function(hakEnv, moduleInfo) {
         });
     }
 
-    // Ensure Rust target exists
+    // Ensure Rust target exists (nb. we avoid depending on rustup)
     await new Promise((resolve, reject) => {
-        childProcess.execFile('rustup', ['target', 'list', '--installed'], (err, out) => {
+        const rustc = childProcess.execFile('rustc', [
+            '--target', hakEnv.getTargetId(), '-o', 'tmp', '-',
+        ], (err, out) => {
             if (err) {
-                reject("Can't find rustup");
+                reject(
+                    "rustc can't build for target " + hakEnv.getTargetId() +
+                    ": ensure the correct toolchain is installed",
```

Comment:
The word "toolchain" here I find a bit confusing... In my mental model of Rust terms, largely guided by the way `rustup` presents them, a "toolchain" is an installation of Rust for a certain host architecture, while a "target" is both the ability to compile for a certain architecture (this part is less important since `rustc --print target-list` shows the huge target list the compiler knows about) and also the SDK for that architecture (this is the part that typically needs some manual action to resolve). The toolchain might match the target in the native case, or it might not when cross-compiling.

```suggestion
                    ": ensure target is installed via `rustup target add " + hakEnv.getTargetId() + "` " +
                    "or your package manager if not using `rustup`",
```

Reply:
Ah yes - for some reason I got 'toolchain' in my head, and then actually did the rustup stuff to get my system aarch64-capable and realised that it was in fact the target that I was talking about.